### PR TITLE
Create discussion field VC-1555

### PR DESCRIPTION
**Pull request by alperbuulut**

### CODE REVIEW

File: public/components/panels/settings/lib/discussion/component.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      pingStatus: pingStatus
+    }
+    settingsStorage.state('commentStatus').set(commentStatus)
+    settingsStorage.state('pingStatus').set(pingStatus)
+
+    this.commentValueChangeHandler = this.commentValueChangeHandler.bind(this)
+    this.pingValueChangeHandler = this.pingValueChangeHandler.bind(this)
+  }
+
+  commentValueChangeHandler (_, value) {
```
Comment (nikitahl) [inline public/components/panels/settings/lib/discussion/component.js:27]: What is the first argument stand for? Maybe it's worth using a more descriptive name, for more readability?

File: public/components/panels/settings/lib/discussion/component.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    if (value) {
+      settingsStorage.state('commentStatus').set('open')
+    } else {
+      settingsStorage.state('commentStatus').set('closed')
+    }
+  }
+
+  pingValueChangeHandler (_, value) {
```
Comment (nikitahl) [inline public/components/panels/settings/lib/discussion/component.js:35]: What is the first argument stand for? Maybe it's worth using a more descriptive name, for more readability?

File: visualcomposer/Modules/Editors/Settings/WordPressSettings/DiscussionController.php [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                            [
+                                'key' => 'VCV_PING_STATUS',
+                                'value' => $pingStatus,
+                            ]
+                        ),
+                    ]
+                );
+            }
+            if (post_type_supports($currentPostType, 'comments')) {
+                $response = array_merge(
```
Comment (cagdasdag) [inline visualcomposer/Modules/Editors/Settings/WordPressSettings/DiscussionController.php:77]: No need to 2 time array_merge. 
Just put if checks and add vcv views to an array and then use array merge

**Review by nikitahl — APPROVED** [review]

File: visualcomposer/Modules/Editors/Settings/WordPressSettings/DiscussionController.php [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+     * @param \VisualComposer\Helpers\Request $requestHelper
+     *
+     * @return mixed
+     */
+    protected function setData($response, $payload, Request $requestHelper)
+    {
+        $postTypeHelper = vchelper('PostType');
+        $currentPost = $postTypeHelper->get();
+        // @codingStandardsIgnoreStart
+        $postCommentStatus = $requestHelper->input('vcv-settings-comment-status', $currentPost->comment_status);
```
Comment (cagdasdag) [inline visualcomposer/Modules/Editors/Settings/WordPressSettings/DiscussionController.php:108]: To make it more secure
use smth like;
$currentPageId = $payload['sourceId'];
$currentPost = $postTypeHelper->get($currentPageId);
$postCommentStatus = $requestHelper->input('vcv-settings-comment-status', $currentPost->comment_status);

File: visualcomposer/Helpers/Localizations.php [hunk 5]
```diff
@@ -1386,6 +1386,18 @@ public function getLocalizations()
                 'Author',
                 'visualcomposer'
             ),
+            'discussionTitle' => __(
+                'Discussion',
```
Comment (FilipsMasolovs) [inline visualcomposer/Helpers/Localizations.php:1390]: Just a question, why are you calling "Discussion" text as "discussionTitle"?
I mean we have "layout" => "Layout", "settings" => "Settings", "update" => "Update".... etc...
So why different here?

File: public/components/panels/settings/lib/discussion/component.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+      pingStatus: pingStatus
+    }
+    settingsStorage.state('commentStatus').set(commentStatus)
+    settingsStorage.state('pingStatus').set(pingStatus)
+
+    this.commentValueChangeHandler = this.commentValueChangeHandler.bind(this)
+    this.pingValueChangeHandler = this.pingValueChangeHandler.bind(this)
+  }
+
+  commentValueChangeHandler (fieldKey, value) {
```
Comment (FilipsMasolovs) [inline public/components/panels/settings/lib/discussion/component.js:27]: Why do you need fieldKey in this function if you are never using it?

File: public/components/panels/settings/lib/discussion/component.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    if (value) {
+      settingsStorage.state('commentStatus').set('open')
+    } else {
+      settingsStorage.state('commentStatus').set('closed')
+    }
+  }
+
+  pingValueChangeHandler (fieldKey, value) {
```
Comment (FilipsMasolovs) [inline public/components/panels/settings/lib/discussion/component.js:35]: Why do you need fieldKey in this function if you are never using it?

**Review by cagdasdag — APPROVED** [review]

**Pull request merged**